src/ray/core_worker/transport/thread_pool.h

```diff
@@ -53,7 +56,7 @@ class BoundedExecutor {
 
  private:
   /// The underlying thread pool for running tasks.
-  boost::asio::thread_pool pool_;
+  std::unique_ptr<boost::asio::thread_pool> pool_;
```

Comment:
context for this change?

Reply:
Because I want to initialize `pool_` after checking the value of `max_concurrency`, I moved its initialization from the initializer list to the constructor.